mat64/dense.go

```diff
@@ -413,21 +431,39 @@ func (m *Dense) Clone(a Matrix) {
 // See the Copier interface for more information.
 func (m *Dense) Copy(a Matrix) (r, c int) {
 	r, c = a.Dims()
+	if a == m {
+		return r, c
+	}
 	r = min(r, m.mat.Rows)
 	c = min(c, m.mat.Cols)
 	if r == 0 || c == 0 {
 		return 0, 0
 	}
 
-	switch a := a.(type) {
+	aU, trans := untranspose(a)
```

Comment:
Note that I've changed docs on Copier to say that aliased data means the behaviour from a call to Copy is undefined.